data/cookbook/extract-links-from-html/00-lambdasoup.ml

```diff
@@ -0,0 +1,59 @@
+---
+packages:
+  - name: "lambdasoup"
+    tested_version: "1.0.0"
+    used_libraries:
+      - lambdasoup
+discussion: |
+  - **Refernce:** The lambdasoup package provides a robust toolset for working with HTML. [github.com/lambdasoup](https://github.com/aantron/lambdasoup?tab=readme-ov-file)
+---
+
+(*
+
+`find_links` accepts an argument `html_content` of type string that contains our HTML content and returns
+the content of the `href` tags.
+
+`parse` from the `Soup` library produces a document node representing the HTML string.
+
+`$$` selects the links in the document.
+
+*)
+open Soup
+
+let find_links html_content =
+  let document_node = Soup.parse html_content in 
```

Comment:
Since we already have `open Soup` above this, `Soup.` prefix is redundant here. Actually we can just do:

```ocaml
parse html_content $$ "a[href]" |> iter ...
```